scipy/optimize/tests/test_linesearch.py

```diff
@@ -142,10 +143,11 @@ def test_scalar_search_wolfe1(self):
 
         assert c > 3  # check that the iterator really works...
 
-    def test_scalar_search_wolfe2(self):
+    @pytest.mark.parametrize("maxiter",[5, 10, 15, 20, 100])
```

Comment:
Instead of adding to a test that already passes on main we should probably add the example from https://github.com/scipy/scipy/issues/17822#issuecomment-1422663435 to demonstrate this fixes the issue

Reply:
Done

Reply:
Sorry I meant adding this as a new test and reverting this change